foo_uie_console/main.cpp

```diff
@@ -406,8 +430,20 @@ LRESULT ConsoleWindow::on_message(HWND wnd, UINT msg, WPARAM wp, LPARAM lp)
         if (s_windows.empty()) {
             DeleteFont(s_font);
             s_font = nullptr;
+            DeleteObject(s_brush);
+            s_brush = nullptr;
         }
         break;
+    case WM_CTLCOLORSTATIC: {
+        HDC hdc = (HDC)wp;
+
+        cui::colours::helper helper(console_colour_id);
+
+        SetTextColor(hdc, helper.get_colour(cui::colours::colour_text));
+        SetBkColor(hdc, helper.get_colour(cui::colours::colour_background));
+
+        return (INT_PTR)s_brush;
```

Comment:
I think the example from Microsoft Docs must be using `INT_PTR` for a dialog box procedure (`DLGPROC`), I'd switch to using a C++ cast to an `LRESULT`:

```suggestion
        return reinterpret_cast<LRESULT>(s_brush);
```